pkg/controller/deployment/sync.go

```diff
@@ -365,17 +365,6 @@ func (dc *DeploymentController) scale(deployment *extensions.Deployment, newRS *
 		return err
 	}
 
-	// If the new replica set is saturated, old replica sets should be fully scaled down.
-	// This case handles replica set adoption during a saturated new replica set.
-	if deploymentutil.IsSaturated(deployment, newRS) {
-		for _, old := range controller.FilterActiveReplicaSets(oldRSs) {
-			if _, _, err := dc.scaleReplicaSetAndRecordEvent(old, 0, deployment); err != nil {
-				return err
-			}
-		}
-		return nil
-	}
```

Comment:
> scaleUpCount is always > 0 because maxTotalPods - currentPodCount > 0 and d.spec.replicas > newrs.spec.replicas.

You mentioned RS adoption earlier. Wouldn't `currentPodCount (# replicas of all RSes) > maxTotalPods (# replicas of Deployment + maxSurge)` when we just adopt a RS from somewhere else? Is this the case you had in mind?